projects/chaos/src/gpio_fsm.c

```diff
@@ -53,21 +86,29 @@ static void prv_gpio_state_idle(FSM *fsm, const Event *e, void *context) {
 static void prv_gpio_state_charge(FSM *fsm, const Event *e, void *context) {
   const ChaosConfig *cfg = context;
   const GPIOAddress sequence[] = {
-    cfg->driver_display_power,
-    cfg->array_sense_power,
-    cfg->battery_box_power,
-    cfg->rear_lights_power,
+    // cfg->charger_power, (external)
+    cfg->driver_display_power, cfg->rear_camera_power,     cfg->array_sense_power,
+    cfg->front_lights_power,   cfg->motor_interface_power,
   };
 
   gpio_seq_set_state(sequence, SIZEOF_ARRAY(sequence), GPIO_STATE_HIGH, GPIO_FSM_SLEW_RATE_US);
 }
 
+static void prv_gpio_state_drive_preconfig(FSM *fsm, const Event *e, void *context) {
```

Comment:
Is the preconfig state necessary? You could probably just set both default and explicit values in the same state output. It's really just setting everything to a known state.

Reply:
I suppose that is true. It's just that I want the preconfig to occur before the battery relay closes and the post config to occur after... That way the DCDCs are started before I attempt to turn stuff on.

Reply:
Yes but this is a matter of ordering from the perspective of the sequencer. I want to intersperse other actions in between.